lib/utils/HttpHelper.dart

```diff
@@ -182,10 +194,17 @@ class HttpHelper {
       'From': OpenFoodAPIConfiguration.getUser(user)?.userId ?? FROM,
     });
 
-    if (isTestModeActive) {
+    if (isTestModeActive && !addCredentialsToHeader) {
       var token = 'Basic ${base64Encode(utf8.encode('off:off'))}';
       headers.addAll({'authorization': token});
     }
+
+    if (addCredentialsToHeader) {
+      final String? userId = OpenFoodAPIConfiguration.getUser(user)?.userId;
+      final String? password = OpenFoodAPIConfiguration.getUser(user)?.password;
+      var token = 'basic ${base64Encode(utf8.encode('$userId:$password'))}';
+      headers.addAll({'Authorization': token});
```

Comment:
I don't know who's right but just a dozen line upwards you spell it `Basic` and `authorization`. Please check the correct spellings and be coherent.

Reply:
From https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization

> The [Authentication scheme](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes) that defines how the credentials are encoded. Some of the more common types are (case-insensitive): [Basic](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#basic_authentication_scheme), Digest, Negotiate and AWS4-HMAC-SHA256.

but I guess we should go with the more used titlecase `Basic`

Reply:
```suggestion
      var token = 'Basic ${base64Encode(utf8.encode('$userId:$password'))}';
      headers.addAll({'Authorization': token});
```